src/EntityRegistrantDetection/CallbackWrapper.php

```diff
@@ -0,0 +1,364 @@
+			'core/social-link',
+			'core/tag-cloud',
+			'core/audio',
+			'core/button',
+			'core/buttons',
+			'core/code',
+			'core/column',
+			'core/columns',
+			'core/cover',
+			'core/embed',
+			'core/freeform',
+			'core/gallery',
+			'core/group',
+			'core/heading',
+			'core/html',
+			'core/image',
+			'core/list',
+			'core/media-text',
+			'core/missing',
+			'core/more',
+			'core/nextpage',
+			'core/paragraph',
+			'core/preformatted',
+			'core/pullquote',
+			'core/quote',
+			'core/separator',
+			'core/social-links',
+			'core/spacer',
+			'core/table',
+			'core/text-columns',
+			'core/verse',
+			'core/video',
+		],
+		'shortcode' => [
+			'caption',
+			'gallery',
+			'audio',
+			'video',
+			'playlist',
+			'embed',
+		],
+	];
+
+	/**
+	 * Callback data.
+	 *
+	 * @var array
+	 */
+	protected $callback;
+
+	/**
+	 * Source of the function.
+	 *
+	 * @var array
+	 */
+	protected $source;
+
+	/**
+	 * List of registered entities by callback.
+	 *
+	 * @var array [
+	 *     @type array $post_type List of registered post types slug.
+	 *     @type array $taxonomy  List of registered taxonomy slug.
+	 *     @type array $block     List of registered block type.
+	 *     @type array $shortcode List of registered shortcode.
+	 * ]
+	 */
+	protected $registered_entities;
+
+	/**
+	 * Instance of EntityRegistrantDetectionManager.
+	 *
+	 * @var EntityRegistrantDetectionManager
+	 */
+	protected $detection_manager;
+
+	/**
+	 * Callback reflection.
+	 *
+	 * @var CallbackReflection
+	 */
+	protected $callback_reflection;
+
+	/**
+	 * AMP_Validation_Callback_Wrapper constructor.
+	 *
+	 * @param EntityRegistrantDetectionManager $detection_manager   Instance of EntityRegistrantDetectionManager.
+	 * @param CallbackReflection               $callback_reflection Instance of CallbackReflection.
+	 * @param array                            $callback            [
+	 *     The callback data.
+	 *     @type callable $function
+	 *     @type int      $accepted_args
+	 *     @type int      $priority
+	 *     @type string   $hook
+	 * ]
+	 */
+	public function __construct( EntityRegistrantDetectionManager $detection_manager, CallbackReflection $callback_reflection, $callback ) {
+
+		$this->detection_manager   = $detection_manager;
+		$this->callback_reflection = $callback_reflection;
+		$this->callback            = $callback;
+		$this->registered_entities = [
+			'post_type' => [],
+			'taxonomy'  => [],
+			'block'     => [],
+			'shortcode' => [],
+		];
+	}
+
+	/**
+	 * Get callback function.
+	 *
+	 * @return callable
+	 */
+	public function get_callback_function() {
+
+		return $this->callback['function'];
+	}
+
+	/**
+	 * Invoke wrapped callback.
+	 *
+	 * @param array ...$args Args.
+	 *
+	 * @return mixed Response.
+	 */
+	public function __invoke( ...$args ) {
+
+		$this->prepare();
+
+		$result = call_user_func_array(
+			$this->get_callback_function(),
+			array_slice( $args, 0, (int) $this->callback['accepted_args'] )
+		);
```

Comment:
Is there a reason why the outdated and slow `call_user_func_array()` is used here?
This could be rewritten to use a faster variadic arguments call:
```suggestion
		$callback = $this->get_callback_function();
		$result   = $callback( ... array_slice( $args, 0, (int) $this->callback['accepted_args'] ));
```
Note that it cannot be done as a one-liner because of PHP 5.6 parsing.